plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/sessionconfig/CodeScanSessionConfig.kt

```diff
@@ -175,39 +180,53 @@ class CodeScanSessionConfig(
 
     fun getProjectPayloadMetadata(): PayloadMetadata {
         val files = mutableSetOf<String>()
+        val traversedDirectories = mutableSetOf<VirtualFile>()
         val stack = Stack<VirtualFile>()
         var currentTotalFileSize = 0L
         var currentTotalLines = 0L
         val languageCounts = mutableMapOf<CodeWhispererProgrammingLanguage, Int>()
 
-        stack.push(projectRoot)
-        while (stack.isNotEmpty()) {
-            val current = stack.pop()
-
-            if (!current.isDirectory) {
-                if (runBlocking { !featureDevSessionContext.ignoreFile(current, this) }) {
-                    if (willExceedPayloadLimit(currentTotalFileSize, current.length)) {
-                        break
+        project.modules.forEach { module ->
+            val changeListManager = ChangeListManager.getInstance(module.project)
+            if (module.guessModuleDir() != null) {
+                stack.push(module.guessModuleDir())
+                while (stack.isNotEmpty()) {
+                    val current = stack.pop()
+
+                    if (!current.isDirectory) {
+                        // Using both featureDevSessionContext and changeListManager to ignore files.
+                        // featureDevSessionContext is used to only ignore files in gitignore in project root and uses regex pattern matching.
+                        if (runBlocking { !featureDevSessionContext.ignoreFile(current, this) } &&
+                            !changeListManager.isIgnoredFile(current) && !files.contains(current.path)
+                        ) {
+                            if (willExceedPayloadLimit(currentTotalFileSize, current.length)) {
+                                break
```

Comment:
this breaks the inner while loop, but not the modules.forEach loop

Reply:
Thanks for the catch.